presto-main/src/main/java/com/facebook/presto/operator/PartitionedLookupSourceFactory.java

```diff
@@ -133,12 +133,23 @@ else if (outer) {
 
         if (lookupSourceSupplier != null) {
             for (SettableFuture<LookupSource> lookupSourceFuture : lookupSourceFutures) {
-                lookupSourceFuture.set(lookupSourceSupplier.get());
+                lookupSourceFuture.set(lookupSourceSupplier.getLookupSource());
             }
         }
     }
 
     @Override
+    public OuterPositionIterator getOuterPositionIterator()
+    {
+        TrackingLookupSourceSupplier lookupSourceSupplier;
+        synchronized (this) {
+            checkState(this.lookupSourceSupplier != null, "lookup source not ready yet");
+            lookupSourceSupplier = this.lookupSourceSupplier;
+        }
+        return lookupSourceSupplier.getOuterPositionIterator();
```

Comment:
Why not just call
`this.lookupSourceSupplier.getOuterPositionIterator()` from synchronized block.